js/src/wallet/parseURL.test.ts

```diff
@@ -0,0 +1,92 @@
+import { PublicKey } from '@solana/web3.js';
+import BigNumber from 'bignumber.js';
+import { parseURL } from './parseURL';
+
+describe('parseURL', () => {
+    describe('parsing', () => {
+        describe('when given correct params', () => {
+            it('should parse successfully', () => {
+                const url =
+                    'solana:mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN?amount=0.01&reference=82ZJ7nbGpixjeDCmEhUcmwXYfvurzAgGdtSMuHnUgyny&label=Michael&message=Thanks%20for%20all%20the%20fish&memo=OrderId5678';
+
+                const parsed = parseURL(url);
+
+                expect(parsed.recipient).toEqual(new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN'));
+                expect(parsed.amount).toEqual(new BigNumber(0.01));
+                expect(parsed.token).toBeUndefined();
+                expect(parsed.references).toEqual([new PublicKey('82ZJ7nbGpixjeDCmEhUcmwXYfvurzAgGdtSMuHnUgyny')]);
```

Comment:
Hmm this is interesting that this works -- how are the two `PublicKey` instances tested for equality here?

Reply:
`.toEqual` does a deep comparison. Loops through each key and compares the values

Reply:
Got it, I just didn't realize `toEqual` automatically called that. Thanks!

Reply:
> Use .toEqual when you want to check that two objects have the same value. This matcher recursively checks the equality of all fields, rather than checking for object identity—this is also known as "deep equal"